lib/Extension/LanguageServerRename/Adapter/ReferenceFinder/ClassMover/ClassRenamer.php

```diff
@@ -81,7 +81,8 @@ public function rename(TextDocument $textDocument, ByteOffset $offset, string $n
 
         $originalName = $this->getFullName($node);
         $newName = $this->createNewName($originalName, $newName);
-        $newUri = TextDocumentUri::fromString($this->nameToUriConverter->convert($newName));
+        $oldUri = $this->nameToUriConverter->convert($originalName);
```

Comment:
it would probably be better to use the refector? `ClassReflector` as that that's a better indicator of where the class is defined:

```
try {
    $path = $this->reflector->reflectClass($originalName)->sourceCode()->path());
} catch (NotFound $notFound) {
   
   // do something
}
if ($path === null) {
   // do somethinh
}

$oldUri = TextDocumentUri::fromString($path);
```

Reply:
This reflector seems to be free from the mentioned bug but it cannot reflect anything nonexisting yet. `$newUri` cannot be obtained this way.

Reply:
yeah, you can only use it for the old one, but I thought that was where the issue was?

Reply:
Yes, it was the issue.